sched/signal/sig_action.c

```diff
@@ -407,11 +447,65 @@ int sigaction(int signo, FAR const struct sigaction *act,
 
 void nxsig_release_action(FAR sigactq_t *sigact)
 {
+  FAR sq_queue_t * const list = &g_sigfreeaction;
+  FAR sigactq_t *prev;
+  FAR sigactq_t *next;
   irqstate_t flags;
+  int i;
 
-  /* Just put it back on the free list */
+  /* Put back to the free list */
+
+  nxsig_add_free(sigact);
```

Comment:
what's benefit to reclaim the signal action pool? if it's important to safe the memory, it's simpler and fast to alloc/free one action from heap.